project.md

```diff
@@ -5,47 +5,65 @@
 
 _What need is met by your idea? Who are you helping? What is that person's
 experience like now? What would their experience be like if you could help
 them?_
 
+	I'm aiming to create a program that would let them quickly prototype things, without having to go through all the hurdles of designing things in a full music production application.
+
 ### Why a language?
 
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+	A language lets you design these fast, in a way that's easy to edit, and in a way that you can easily export to different forms using existing libraries. This helps to make it a quick way to prototype melodies and such, rather than having to use a graphical representation of notes that may be slower. (built in repetition and ways to vary it, especially, should be an good way to specificaully work towards this need.)
+
 ### Why you?
 
 _What excites you about this idea? How did you come up with it?_
 
+	I came up with it as a good project that allows me to design something novel, with unique skills added. It also lets me design a project about a domain I already have a little bit of experience in.
+
 ### Domain
 
 _Describe the project's domain in five words._
 
+	Producing sheet music or melodies
+
 ### Interface (syntax)
 
 _How might the user interact with the language? What does programming look
 like? Why is this the right way to interact with the problem domain?_
 
+	A user interacts with this language by writing a new song in plaintext. The user would define a number of chords and melodies as a collection of notes, then would put together these chords and melodies in increasinly complicated ways to eventually lead to a final song. This lets you define a structure and then use it repeatedly, which should make it quicker to edit structures used across the program.
+
 ### Operation (semantics)
 
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+	When the program runs, it'll parse the plaintext file into labels for defined sets of notes, and output commands. When it comes across an output command (like creating a sheet music image, or sound file), It'll add all the given notes into that output file-whenever it comes across a label, it'll replace that with the notes it's comprised of, then modified by things like repeating some melody.
+
 ### Expressiveness
 
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+	It should be easily possible to define a melody or chord quickly, and it should also be possible to put those together in order to make a full 'song.' Ideally, it should be easy to take a set of notes and modify it in some way- shift it's key or repeat it or cut it short. It should be possible, but difficult, to design an entire song that's not built up out of smaller parts.
```

Comment:
I can definitely see how this would be helpful for "supporting" staffs. For example, the left hand of a piano tends to follow repeating patterns while the right hand will often undergo more complex melodies.